log/examples.md

```diff
@@ -0,0 +1,139 @@
+# Examples of Usage
+
+## Setup
+
+```go
+package main
+
+import (
+    "context"
+
+    "github.com/anz-bank/pkg/log"
+    "github.com/anz-bank/pkg/log/loggers"
+)
+
+func main() {
+    // User is expected to choose a logger and add it to the context using the library's API
+    ctx := context.Background()
+
+    // this is a logger based on the logrus standard logger
+    logger := loggers.NewStandardLogger()
+
+    // WithLogger returns a new context
+    ctx = log.WithLogger(ctx, logger)
+}
+```
+
+That's all in setup, now logging can be used by using the context.
+
+## Usage
+
+```go
+import (
+    "github.com/anz-bank/pkg/log"
+)
+
+func stuffToLog(ctx context.Context) {
+    // logging requires the context variable so it must be given to any function that requires it
+    log.Logger(ctx).Debug("Debug")
+    log.Logger(ctx).Print("Print")
+    log.Logger(ctx).Trace("Trace")
+    log.Logger(ctx).Warn("Warn")
+    log.Logger(ctx).Error("Error")
+    log.Logger(ctx).Fatal("Fatal")
+    log.Logger(ctx).Panic("Panic")
+
+    // Expected to log
+    // (time in RFC3339Nano Format) (Level) (Message)
+    //
+    // Example:
+    // 2019-12-12T08:23:59.210878+11:00 PRINT Hello There
+    //
+    // Each API also has its Format counterpart (Debugf, Printf, Tracef, Warnf, Errorf, Fatalf, Panicf)
+}
+```
+
+Fields are also supported in the logging. There are two kinds of fields, context-level fields and log-level fields.
+
+```go
+import (
+    "github.com/anz-bank/pkg/log"
+)
+
+
+// With fields, it is expected to log
+// (time in RFC3339Nano Format) (Fields) (Level) (Message)
+//
+// Fields will be logged ALPHABETICALLY. If the same key field is added to the context logger,
```

Comment:
where is this guarenteed/implemented in the code?

Reply:
https://github.com/nofun97/pkg/blob/logger-lib/log/loggers/standardLogger.go#L151

Reply:
i added the comment in the test